### PR TITLE
[CORE-1626] - Show BlackBook vehicle specification PDF on appraisal details views

### PR DESCRIPTION
**Pull request by pabloarf**
**- What I did**
Added functionality needed by web-system to get Blackbook PDFs with vehicle specs.

**- How I did it**
Added functionality to get UVC from vehicle by VIN.
Added specs.
Did refactor of the rest_client and vin_service. Refactor pending to do in the other services.

**- How to verify it**
Run the specs.

**- Description for the changelog**
Show BlackBook vehicle specification PDF on appraisal details views

**- A picture of a cute animal (not mandatory but encouraged)**
description

### CODE REVIEW

File: lib/mappers/range_values_mapper.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module RangeValuesMapper
+  def self.map_fields(response)
+    return {} unless response && response['used_vehicles'] && response['used_vehicles']['used_vehicle_list']
```
Comment (ebordon) [inline lib/mappers/range_values_mapper.rb]: Maybe we can use [dig](https://ruby-doc.org/core-2.3.0/Hash.html#method-i-dig) to this kind of statements: `return {} unless response.dig('used_vehicles', 'used_vehicle_list')`

File: lib/mappers/uvc_mapper.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module UvcMapper
+  def self.map_fields(response)
+    return {} unless response && response['used_vehicles'] && response['used_vehicles']['used_vehicle_list']
```
Comment (ebordon) [inline lib/mappers/uvc_mapper.rb]: Same here

Comment (ebordon) [reply lib/mappers/uvc_mapper.rb]: Also, this map_fields method seems to be exactly the same as the one of RangeValuesMapper. We may think in a common method or some class inheritance.

File: lib/rest_client.rb [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
     { Authorization: "Basic #{token}" }
   end
 
+  def make_request(method, url, query_params = {})
+    return unless [:get].include?(method)
+    begin
+    response = JSON.parse(HTTParty.send( method, url, headers: headers, query: query_params))
+    status = 200
+
+    if response['error_count'].positive?
```
Comment (ebordon) [inline lib/rest_client.rb]: Should we check `response.status` before checking the data?

File: lib/rest_client.rb [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+      response = { error: response['message_list'][0]['description'] }
+      status = 400
+    end
+
+    [response, status]
```
Comment (ebordon) [inline lib/rest_client.rb]: I prefer to return a hash instead an array
```
{ status: status,
  response: response
}
```

File: lib/vin_service.rb [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
 
-    begin
-      response = HTTParty.get(url, headers: headers, query: query_params)
-      process_response(JSON.parse(response), :map_fields)
-    rescue => e
-      [{ error: e.to_s }, 500]
-    end
+    response = make_request(:get, url, query_params)
+    return response unless response.last == 200
+    [RangeValuesMapper.map_fields(response.first), response.last]
```
Comment (ebordon) [inline lib/vin_service.rb]: Why not return a hash?

File: lib/rest_client.rb (hunk 4 above)
Comment (pabloarf) [reply lib/rest_client.rb]: For now, chose to return an array so this changes are backwards compatible

File: lib/blackbook_client/version.rb [hunk 6]
```diff
@@ -1,3 +1,3 @@
 class BlackbookClient
-  VERSION = '0.1.6'
+  VERSION = '0.2.0'
```
Comment (cschmeichel) [inline lib/blackbook_client/version.rb]: You probably forgot to run `bundle`. I don't see changes in the gemfile.lock

File: lib/mappers/spec_pdf_mapper.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+module SpecPdfMapper
+  def self.map_fields(response)
+    return {} unless response&.dig('spec_pdf', 'file_contents')
+    file_contents = response['spec_pdf']['file_contents']
```
Comment (cschmeichel) [inline lib/mappers/spec_pdf_mapper.rb]: you could combine these two lines doing:
`return {} unless file_contents = response['spec_pdf']['file_contents']`

File: lib/rest_client.rb [hunk 8]
```diff
@@ -1,11 +1,36 @@
...
   def headers
     token = Base64.encode64("#{username}:#{password}")
     { Authorization: "Basic #{token}" }
   end
 
+  def make_request(method, url, query_params = {})
+    return unless [:get].include?(method)
+    begin
+    response = JSON.parse(HTTParty.send(method, url, headers: headers, query: query_params))
+    status = 200
```
Comment (cschmeichel) [inline lib/rest_client.rb]: Why are you forcing the status 200?

File: lib/rest_client.rb [hunk 9]
```diff
@@ -1,11 +1,36 @@
...
 
   def headers
     token = Base64.encode64("#{username}:#{password}")
     { Authorization: "Basic #{token}" }
   end
 
+  def make_request(method, url, query_params = {})
+    return unless [:get].include?(method)
+    begin
+    response = JSON.parse(HTTParty.send(method, url, headers: headers, query: query_params))
```
Comment (cschmeichel) [inline lib/rest_client.rb]: You probably need to rescue some errors here, right?

File: lib/blackbook_client/version.rb (hunk 6 above)
Comment (pabloarf) [reply lib/blackbook_client/version.rb]: Solved! 👍🏻

**Pull request merged**